test/lib/ansible_test/_internal/provider/layout/__init__.py

```diff
@@ -169,6 +170,19 @@ def __init__(self,
         self.full_name = '%s.%s' % (namespace, name)
         self.prefix = '%s.' % self.full_name
         self.directory = os.path.join('ansible_collections', namespace, name)
+        self.version = None
+
+        try:
+            import yaml
+        except ImportError:
+            raise ApplicationError('Need PyYAML to load galaxy.yml. Please make sure that PyYAML has been installed.')
+        try:
+            with open(os.path.join(self.root, self.directory, 'galaxy.yml')) as galaxy_file:
+                galaxy = yaml.safe_load(galaxy_file)
+        except IOError as exc:
+            raise ApplicationError('Error while reading galaxy.yml: {0}'.format(exc))
+
+        self.version = galaxy.get('version')
```

Comment:
This should work on both source and published collections, which means it will need to support reading from both `galaxy.yml` and `MANIFEST.json`. If the version lookup fails for any reason, it should continue with a warning. When the pylint plugin or validate-modules attempt to check a deprecated version, at that point they should generate an error if the version was not provided.

This will ensure that `ansible-test` can function on a local collection with no version, or corrupted metadata. The only tests which will report errors are the ones trying to use that version information at the time of use. If no version checking needs to be done, lack of a version to check against shouldn't be an error.

Also, the version lookup in `ansible-test` itself should be lazy-loaded, and deferred to a subprocess.

For an example, see:

https://github.com/ansible/ansible/blob/0651e1c25fab26c33130128774772444957fc524/test/lib/ansible_test/_internal/ansible_util.py#L190

That way `ansible-test` can be run from source, such as from GitHub, to test a collection using delegation (`--docker` for example) and not require `yaml` to be installed to avoid a spurious warning. It also makes sure that the correct python is used to import `yaml` when using the `--python` option, as that interpreter may be different than the one used to run `ansible-test`.

Reply:
I've added support for MANIFEST.json in 3e51422b049453b23b9c2fb668c912f006514d18, and then moved metadata collection to a subprocess in eea0120ecfafa9ae563fc99688b6c590889c8074 as you suggested. I put the resulting function also in ansible_util.py. Do you prefer another place?

BTW, I noticed that `ansible-test sanity` doesn't really works in galaxy-installed collections. I always get `WARNING: All targets skipped.` and no test is run. Happens both with this branch and with `devel`.

Reply:
@felixfontein Can you open an issue for that?

Reply:
Done (#68499)